static/routing-tree.js

```diff
@@ -118,6 +118,9 @@ function matchLabel(matcher, labelSet) {
   var v = labelSet[matcher.name];
 
   if (matcher.isRegex) {
+    // Prevent regex matches on undefined labels.
+    if (undefined === v) return false;
```

Comment:
Rather than return false, I'd change "undefined" to an empty string.

For example if 'app: .+' was changed to 'app: .*', matching an empty label value would result in no route to B being drawn.

Reply:
`v` is now rewritten to an empty string